upt_macports/upt_macports.py

```diff
@@ -30,9 +30,15 @@ def licenses(self):
         filepath = pkg_resources.resource_filename(__name__, relpath)
         with open(filepath) as f:
             spdx2macports = json.loads(f.read())
-
-        return ' '.join([spdx2macports.get(license.spdx_identifier, 'unknown')
-                        for license in self.upt_pkg.licenses]) or 'unknown'
+        licenses = []
+        for license in self.upt_pkg.licenses:
+            port_license = spdx2macports.get(license.spdx_identifier,
+                                             'unknown')
+            if license.spdx_identifier != 'unknown'and port_license == 'unknown': # noqa
+                self.logger.error(f'spdx2macports license conversion failed for {license.spdx_identifier}') # noqa
```

Comment:
Why the "# noqa" on these lines?

Also, is this really an error? Shouldn't it rather be a warning?

Reply:
for the second line, it would be necessary I guess, else the indentation would look very weird.

I was confused about error vs warning but went with error as it is something that should be fixed and if we go with a warning, people would tend to ignore it.

Reply:
Error kind of gives the idea that something literally failed (even though we have ".critical()" also). What are the users supposed to do about this warning?

Isn't this more readable? 

```python
        licenses = []
        for license in self.upt_pkg.licenses:
            try:
                if license.spdx_identifier == 'unknown':
                    port_license = 'unknown'
                else:
                    port_license = spdx2macports.get[license.spdx_identifier]
                licenses.append(port_license)
            except KeyError:
                err = f'Unknown license {license.spdx_identifier}'
                self.logger.error(err)
        return ' '.join(licenses) or 'unknown'

```